### PR TITLE
Read lfp

### PR DESCRIPTION
**Pull request by Paul-Baudin**
Some small modifications about readLFP.m :

  -  With the same logic that you use ‘continue’ in for loops, I think ‘return’ can improve readability when use with the ‘force’ variable (avoiding an ‘if’ statement for nearly the whole file).
 -   It is nice for how I deal with dtx experiments (where I have to load LFP) vs control experiment (where I do not load LFP for now), to be able to launch readLFP even if there are no LFP data to load (ie cfg.LFP.name is empty), and make it return and empty array. For that I need to initialise LFP array at the begining of the file, so it is outputed if nothing else is done.
 -   I corrected the EMG part as you asked. The ft_getopt is very helpful to make a clearer code for data with different needs in cfg fields.
 -   I added fsample in the outputed LFP as I think it is easier to do it here where the header is already loaded, whereas to re-search for it later
 -   I added some trialinfos columns. It allows me to easier find later where the trials are located in the initial neuralynx file. I use it to search for trials which intersect BAD markers much more easily : no need to re_use cfg.muse.startend, and, find trials location regardless of how the trials where defined (see removetrials_MuseMarkers.m in a later pull request).

Let me know if you have any improvement or better ideas for this :)

### CODE REVIEW

File: shared/readLFP.m [hunk 1]
```diff
@@ -55,13 +55,18 @@
...
     fprintf('********************************\n');
     fprintf('*** (re-) computing LFP data ***\n');
     fprintf('********************************\n\n');
+end
     
     % get file format 
     [isNeuralynx, isMicromed, isBrainvision] = get_data_format(cfg);
+    
+    % initialize LFP, to return empty array in case there is no LFP to load
+    LFP = [];
```
Comment (stephenwhitmarsh) [inline shared/readLFP.m]: Should be an empty struct {} and not an empty array [], right?

File: shared/readLFP.m [hunk 2]
```diff
@@ -55,13 +55,18 @@
     fprintf('*** Loading precomputed LFP data ***\n');
     fprintf('************************************\n\n');
     load(fname_out,'LFP');
+    return
```
Comment (stephenwhitmarsh) [inline shared/readLFP.m:58]: Very nice! didn't spot that one, will do it in the other functions as well every time a new one is committed.

File: shared/writeSpykingCircusParameters.m [hunk 3]
```diff
@@ -26,7 +26,7 @@ function writeSpykingCircusParameters(cfg)
 cfg.circus.part_list                = ft_getopt(cfg.circus, 'part_list', 'all');
 
 if strcmp(cfg.circus.part_list, 'all')
-    cfg.circus.part_list = 1:size(MuseStruct,2);
+    cfg.circus.part_list = 1:size(cfg.directorylist,2);
```
Comment (stephenwhitmarsh) [inline shared/writeSpykingCircusParameters.m:29]: yes, I was doing that in some other functions as well

File: shared/readLFP.m [hunk 4]
```diff
@@ -250,6 +248,9 @@
                     cfgtemp.trl(:,4)                = trialnr;
                     cfgtemp.trl(:,6)                = idir;
                     cfgtemp.trl(:,7)                = Stage;
+                    cfgtemp.trl(:,8)                = Startsample;
```
Comment (stephenwhitmarsh) [inline shared/readLFP.m:251]: just FYI: this ends up in data.sampleinfo (except offset). 
I was thinking of to deal with these fields though, because they are becoming more, and it becomes harder to make sure which column is which without checking. Now, fieldtrip has started to support MATLAB tables in trialinfo, so perhaps we should add these those (after redefining trials), which will also allow the naming of the fields, rather than just some arbitrary column number.
In short, i'm fine with this, but it will change again soon towards a more sustainable solution.

**Comment by Paul-Baudin** [comment]
Thanks for your feedback. I do not have the sampleinfo field in my LFP data (maybe because of ft_appenddata ?). I totally agree that it would be much better with table so we can find the trialinfo column not with an arbitrary number but with a string name. 
You are right the {} is more consistent with the LFP structure than [], I’ve modified it in this PR.
